src/web/services/account.service.ts

```diff
@@ -0,0 +1,61 @@
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { AccountInfo, AdminSearchResult, JoinLink, MessageOutput } from '../types/api-output';
+import { AccountCreateRequest } from '../types/api-request';
+import { HttpRequestService } from './http-request.service';
+
+/**
+ * Handles account related logic provision
+ */
+
+@Injectable({
+  providedIn: 'root',
+})
+
+export class AccountService {
+  constructor(private httpRequestService: HttpRequestService) {
+  }
+
+  /**
+   * Creates an account by calling API.
+   */
+  createAccount(paramMap: {[key: string]: string}, request: AccountCreateRequest): Observable<JoinLink> {
```

Comment:
Same here, don't rely on the caller to populate the param map. You get the idea, I won't bother for the rest of this file.